grammar.js

```diff
@@ -453,7 +453,7 @@ module.exports = grammar({
     use: ($) =>
       seq(
         "use",
-        optional(series_of($._pattern, ",")),
+        optional(series_of(seq($._pattern, optional($._type_annotation)), ",")),
```

Comment:
I think we should make a `use_assignments` node and a `use_assignment` node (corresponding to the `UseAssignment` node in the Gleam parser) here.  My rationale is just that the current structure is a bit difficult for humans (and likely machines) to understand.  For instance, the following code is valid:

```gleam
use a, b: String, c: Int, d <- foo()
```

The AST generated by `tree-sitter-gleam would look like this:
```scheme
(use
  (identifier)
  (identifier)
  (type
    (type_identifier))
  (identifier)
  (type
    (type_identifier))
  (identifier)
  (function_call
    (identifier)
    (arguments)))
```

To me, it's a bit hard to see what's going on.  The alternative would be something like this:
```suggestion
        optional(field("assignments", $.use_assignments), ",")),
```

yeilding this:

```scheme
(use
  (use_assignments
    (use_assignment
      (identifier))
    (use_assignment
      (identifier)
      (type
        (type_identifier)))
    (use_assignment
      (identifier)
      (type
        (type_identifier)))
    (use_assignment
      (identifier)))
  (function_call
    (identifier)
    (arguments)))
```

Notably the Gleam parser has no `UseAssignments` node, but then again the Gleam parser has no `FunctionParameters` node either but we have a `function_parameters` node; we have to compensate for not really having "lists".

Reply:
Hi! Thanks for reflecting on this!
I am very n00b about how to do tree-sitter grammars and I suppose what you propose is absolutely a good idea!